semantic-core/src/Core/Name.hs

```diff
@@ -1,25 +1,34 @@
-{-# LANGUAGE DeriveGeneric, DeriveTraversable, GeneralizedNewtypeDeriving, LambdaCase, OverloadedLists #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
 module Core.Name
-( module Analysis.Name
+( module Analysis.Functor.Named
 , reservedNames
 , isSimpleCharacter
 , needsQuotation
 ) where
 
-import           Analysis.Name
+import           Analysis.Functor.Named
 import qualified Data.Char as Char
+import           Data.Hashable
 import           Data.HashSet (HashSet)
 import qualified Data.HashSet as HashSet
-import           Data.Text as Text (any, unpack)
+import           Data.String
+import           Data.Text as Text (any)
 
-reservedNames :: HashSet String
+reservedNames :: (Eq s, IsString s, Hashable s) => HashSet s
```

Comment:
This lets us use `reservedNames` in both the lexer and the pretty-printer without any bridging between Name/Text/String.

Reply:
Nice!